core/src/main/java/com/amplifyframework/core/model/AuthStrategy.java

```diff
@@ -25,23 +25,62 @@
      * Owner authorization specifies whether a user can access or operate against an object.  To use OWNER, the API
      * must have Cognito User Pool configured.
      */
-    OWNER,
+    OWNER(Provider.userPools),
 
     /**
      * Group authorization specifies whether a group can access or operate against an object.  To use GROUPS, the API
      * must have Cognito User Pool configured.
      */
-    GROUPS,
+    GROUPS(Provider.userPools),
 
     /**
      * The private authorization specifies that everyone will be allowed to access the API with a valid JWT token from
      * the configured Cognito User Pool. To use PRIVATE, the API must have Cognito User Pool configured.
      */
-    PRIVATE,
+    PRIVATE(Provider.userPools),
 
     /**
      * The public authorization specifies that everyone will be allowed to access the API, behind the scenes the API
      * will be protected with an API Key. To use PUBLIC, the API must have API Key configured.
      */
-    PUBLIC,
+    PUBLIC(Provider.apiKey);
+
+    private final Provider defaultAuthProvider;
+
+    AuthStrategy(Provider defaultAuthProvider) {
+        this.defaultAuthProvider = defaultAuthProvider;
+    }
+
+    /**
+     * Returns the default provider for the strategy.
+     * @return The default provider for the strategy.
+     */
+    public Provider getDefaultAuthProvider() {
+        return defaultAuthProvider;
+    }
+
+    /**
+     * Represents the the value of the provider field of the @auth directive.
+     */
+    public enum Provider {
+        /**
+         * The userPools provider of the @auth rule directive.
+         */
+        userPools,
```

Comment:
nit: as much as I hate how ugly it becomes, the Java convention is to make them all caps since they are constants

Reply:
Yeah...I was torn on this one simply because I wanted to keep it similar to the values that they are actually representing, but I agree with you. Probably more idiomatic to use upper case here.